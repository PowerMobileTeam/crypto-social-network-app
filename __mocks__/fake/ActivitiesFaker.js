
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

