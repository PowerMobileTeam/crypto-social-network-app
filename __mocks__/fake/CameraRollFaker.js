
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

