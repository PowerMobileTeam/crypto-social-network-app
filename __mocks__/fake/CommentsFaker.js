
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

