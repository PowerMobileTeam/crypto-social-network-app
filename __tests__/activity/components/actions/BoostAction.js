import 'react-native';
import React from 'react';
import { Text, TouchableHighlight } from "react-native";
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import BoostAction from '../../../../src/newsfeed/activity/actions/BoostAction';

import withPreventDoubleTap from '../../../../src/common/components/PreventDoubleTap';

import featuresService from '../../../../src/common/services/features.service';

// prevent double tap in touchable

describe('Boost action component', () => {

  let screen;
  beforeEach(() => {

    const TouchableHighlightCustom = <TouchableHighlight onPress={this.onPress} />;

    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <BoostAction entity={activityResponse.activities[0]} navigation={navigation} />
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have A boost button', async () => {
    screen.update();
    expect(screen.find('Text').dive().text()).toBe('BOOST');
  });

  it('should navigate to boost on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = { 
      navigate: jest.fn() 
    };
    let entity = activityResponse.activities[0];
    screen = shallow(
      <BoostAction entity={entity} navigation={navigation}/>
    );
    screen.update()
    let render = screen.dive();
    let touchables = render.find('PreventDoubleTap');
    touchables.at(0).props().onPress();
    jest.runAllTimers();

    expect(navigation.navigate).toHaveBeenCalled();

    expect(screen.find('PreventDoubleTap')).toHaveLength(1);

  });

});
