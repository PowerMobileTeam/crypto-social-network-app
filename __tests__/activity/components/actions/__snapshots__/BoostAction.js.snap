// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boost action component renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <PreventDoubleTap
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#4690D6",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 4,
          "padding": 4,
        },
        undefined,
      ]
    }
    underlayColor="transparent"
  >
    <Text
      accessible={true}
      adjustsFontSizeToFit={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": "Roboto",
            "fontSize": 12,
            "paddingLeft": 2,
            "paddingRight": 2,
          },
          Object {
            "color": "#4690D6",
          },
        ]
      }
    >
      BOOST
    </Text>
  </PreventDoubleTap>
</Component>
`;
